tutorials/analyze-cryptocurrency-data.md

```diff
@@ -0,0 +1,624 @@
+17 different fiat currencies since 2010. The Bitcoin table is below and the Ethereum
+table is very similar:
+
+|Field|Description|
+|---|---|
+|`time`|The day-specific timestamp of the price records, with time given as the default 00:00:00+00|
+|`opening_price`|The first price at which the coin was exchanged that day|
+|`highest_price`|The highest price at which the coin was exchanged that day|
+|`lowest_price`|The lowest price at which the coin was exchanged that day|
+|`closing_price`|The last price at which the coin was exchanged that day|
+|`volume_btc`|The volume exchanged in the cryptocurrency value that day, in BTC.|
+|`volume_currency`|The volume exchanged in its converted value for that day, quoted in the corresponding fiat currency.|
+|`currency_code`|Corresponds to the fiat currency used for non-btc prices/volumes.|
+
+Lastly, we have the `currency_info` table, which maps the currency's code to its English-language name:
+
+|Field|Description|
+|---|---|
+|`currency_code`|2-7 character abbreviation for currency. Used in other hypertables|
+|`currency`|English name of currency|
+
+Once we’ve established the schema for the tables in our database, we can formulate 
+`create_table` SQL statements to actually create the tables we need:
+
+```python
+--Schema for cryptocurrency analysis
+DROP TABLE IF EXISTS "currency_info";
+CREATE TABLE "currency_info"(
+   currency_code   VARCHAR (10),
+   currency        TEXT
+);
+
+--Schema for btc_prices table
+DROP TABLE IF EXISTS "btc_prices";
+CREATE TABLE "btc_prices"(
+   time            TIMESTAMP WITH TIME ZONE NOT NULL,
+   opening_price   DOUBLE PRECISION,
+   highest_price   DOUBLE PRECISION,
+   lowest_price    DOUBLE PRECISION,
+   closing_price   DOUBLE PRECISION,
+   volume_btc      DOUBLE PRECISION,
+   volume_currency DOUBLE PRECISION,
+   currency_code   VARCHAR (10)
+);
+
+--Schema for crypto_prices table
+DROP TABLE IF EXISTS "crypto_prices";
+CREATE TABLE "crypto_prices"(
+   time            TIMESTAMP WITH TIME ZONE NOT NULL,
+   opening_price   DOUBLE PRECISION,
+   highest_price   DOUBLE PRECISION,
+   lowest_price    DOUBLE PRECISION,
+   closing_price   DOUBLE PRECISION,
+   volume_crypto   DOUBLE PRECISION,
+   volume_btc      DOUBLE PRECISION,
+   currency_code   VARCHAR (10)
+);
+
+--Schema for eth_prices table
+DROP TABLE IF EXISTS "eth_prices";
+CREATE TABLE "eth_prices"(
+   time            TIMESTAMP WITH TIME ZONE NOT NULL,
+   opening_price   DOUBLE PRECISION,
+   highest_price   DOUBLE PRECISION,
+   lowest_price    DOUBLE PRECISION,
+   closing_price   DOUBLE PRECISION,
+   volume_eth      DOUBLE PRECISION,
+   volume_currency DOUBLE PRECISION,
+   currency_code   VARCHAR (10)
+);
+
+--Timescale specific statements to create hypertables for better performance
+SELECT create_hypertable('btc_prices', 'time', 'opening_price', 2);
+SELECT create_hypertable('eth_prices', 'time', 'opening_price', 2);
+SELECT create_hypertable('crypto_prices', 'time', 'currency_code', 2);
```

Comment:
```suggestion
SELECT create_hypertable('btc_prices', 'time');
SELECT create_hypertable('eth_prices', 'time');
SELECT create_hypertable('crypto_prices', 'time');
```

Per @svenklemm suggestion, probably good to just stick with time partitioning for simplicity.